packages/jest-runner/src/runTest.ts

```diff
@@ -104,16 +106,46 @@ async function runTestInternal(
     });
   }
 
+  const Runtime: typeof RuntimeClass = config.moduleLoader
+    ? require(config.moduleLoader)
+    : require('jest-runtime');
+  const transformer = new Runtime.ScriptTransformer(config);
+
+  transformer.preloadTransformer(testEnvironment);
+
+  let transforming = false;
+  const revertHook = addHook(
+    (code, filename) => {
+      try {
+        transforming = true;
+        return transformer.transformSource(filename, code, false).code || code;
+      } finally {
+        transforming = false;
+      }
+    },
+    {
+      exts: [extname(testEnvironment)],
+      ignoreNodeModules: false,
+      matcher: (...args) => {
+        if (transforming) {
+          // Don't transform any dependency required by the transformer itself
+          return false;
+        }
+        return transformer.shouldTransform(...args);
+      },
+    },
+  );
+
+  const requireEnvironment = require(testEnvironment);
   const TestEnvironment: typeof JestEnvironment = interopRequireDefault(
-    require(testEnvironment),
```

Comment:
The require override was not working properly when `require` was called within `interopRequireDefault`. So had to come up with a work around of storing it in a variable

Reply:
This was resolved when I extracted the transform code to `jest-transform`